docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,112 @@
+# TileEntities
+
+This is often used by vanilla Minecraft to handle inventories on chests, smelting logic on furances, or area effects for beacons. 
+More advanced versions exist in mods such as quaries, sorting machines, pipes, and displays
+!!! note
+	`TileEntities` aren't a solution for everything, their loading distance is limited, and they can cause lag when used in large amounts.
+	When possible, try to avoid them.
+
+## Creating a `TileEntity`
+In order to create a `TileEntity` you need to extend the `TileEntity` class.
+It is important that your `TileEntity` has a default constructor, so that Minecraft can properly load it.
+After you've created your class, you need to register the `TileEntity`. For this you need to call 
+```JAVA
+	GamerRegistry#registerTileEntity(Class<? extends TileEntity> tileEntityClass, ResourceLocation key)
+```
+Where the first parameter is simply your TileEntity class and the second the registry name of your TileEntity.
+
+!!! note
+	The method to register a `TileEntity` is using a `String` instead of a `ResourceLocation` before Forge version `14.23.3.2694`
+	You should still use the `ResourceLocation` format `modid:tileentity` though.
+	Not using this format will result in your entitiy being called `minecraft:tileentity`
+
+## Attaching a `TileEntity` to a `Block`
+To attach your new `TileEntity` to a `Block` you need to override 2 (two) methods within the Block class.
+```JAVA
+	Block#hasTileEntity(IBlockstate state)
+
+	Block#createTileEntity(World world, IBlockState state)
+```
+Using the parameters you can choose if the block should have a `TileEntity` or not.
+But usually you will return `true` in the first method and a new instance of your `TileEntity` in the second method.
+
+At this point the `TileEntity`can be used for Rendering.
+
+## Storing Data within your `TileEntity`
+In order to save data, create the following two methods
+```JAVA
+	TileEntity#writeToNBT(NBTTagCompound nbt):NBTTagCompound
+
+	TileEntity#readFromNBT(NBTTagCompound nbt)
+```
+This methods are called whenever the `TileEntity` gets (un-)loaded, you don't use them to access your data.
+All your data should be fields within you class, and be (de-)serialized through this methods.
```

Comment:
I would suggest the following:

Use these methods to read and write to the fields in your tile entity class.